### PR TITLE
GitHub Actions: Resolve warnings

### PR DESCRIPTION
**Pull request by batbattur**
## Summary
Updated `actions/checkout` to `v3` and `ramsey/composer-install` to `v2`.

## QA notes
Make sure there are no warnings in the CI check summary.

connects https://github.com/ifixit/ifixit/issues/47011

### CODE REVIEW

**Comment by batbattur** [comment]
dev_block 🚛 WIP

**Comment by batbattur** [comment]
un_dev_block 🚛

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -37,6 +37,8 @@ jobs:
         uses: "ramsey/composer-install@v2"
         with:
           dependency-versions: "${{ matrix.dependencies }}"
+        env:
+          COMPOSER_AUTH: '{"github-oauth": {"github.com": "${{ secrets.GITHUB_TOKEN }}"}}'
```
Comment (ardelato) [inline .github/workflows/ci.yaml:41]: Wait, is this a PAT?

**Review by ardelato — APPROVED** [review]
CR ⚡ but deploy_block ☁️ on a question.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (batbattur) [reply .github/workflows/ci.yaml:41]: I think it comes from Github itself https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret

The permissions for forked repos are all `read` https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

**Comment by ardelato** [comment]
un_deploy_block ⚡

**Comment by jordycosta** [comment]
QA 🟢 

There are no longer warnings regarding _Node.js_ in the CI summary ✔️ 

https://github.com/iFixit/php-ds-polyfill/actions/runs/4702855713/jobs/8340621468?pr=3

**Pull request merged**